reports/template_test.go

```diff
@@ -0,0 +1,376 @@
+// Copyright (c) Abstract Machines
+// SPDX-License-Identifier: Apache-2.0
+
+package reports_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/absmach/magistrala/reports"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
```

Comment:
Same here, instead of writing more than one template, use a valid one and `strings. Replace` to create invalid templates. To add an undefined function, you can still use a similar approach, something like:
```go
unknownFunc := strings.Replace(validTmpl,
	"{{end}}\n\t</body>",
	"<p>Custom: {{customFunction .}}</p>\n{{end}}\t</body>",
	1)
```